test/jdk/java/util/ServiceLoader/BadProvidersTest.java

```diff
@@ -207,55 +210,35 @@ public void testBadProvider(String testName, String ignore) throws Exception {
     public void testWithTwoFactoryMethods() throws Exception {
         Path mods = compileTest(TEST1_MODULE);
 
-        ClassWriter cw = new ClassWriter(ClassWriter.COMPUTE_MAXS
-                                         + ClassWriter.COMPUTE_FRAMES);
-        cw.visit(V9,
-                ACC_PUBLIC + ACC_SUPER,
-                "p/ProviderFactory",
-                null,
-                "java/lang/Object",
-                null);
-
-        // public static p.Service provider()
-        MethodVisitor mv = cw.visitMethod(ACC_PUBLIC + ACC_STATIC,
-                "provider",
-                "()Lp/Service;",
-                null,
-                null);
-        mv.visitTypeInsn(NEW, "p/ProviderFactory$1");
-        mv.visitInsn(DUP);
-        mv.visitMethodInsn(INVOKESPECIAL,
-                "p/ProviderFactory$1",
-                "<init>", "()V",
-                false);
-        mv.visitInsn(ARETURN);
-        mv.visitMaxs(0, 0);
-        mv.visitEnd();
-
-        // public static p.ProviderFactory$1 provider()
-        mv = cw.visitMethod(ACC_PUBLIC + ACC_STATIC,
-                "provider",
-                "()Lp/ProviderFactory$1;",
-                null,
-                null);
-        mv.visitTypeInsn(NEW, "p/ProviderFactory$1");
-        mv.visitInsn(DUP);
-        mv.visitMethodInsn(INVOKESPECIAL,
-                "p/ProviderFactory$1",
-                "<init>",
-                "()V",
-                false);
-        mv.visitInsn(ARETURN);
-        mv.visitMaxs(0, 0);
-        mv.visitEnd();
-
-        cw.visitEnd();
+        var bytes = Classfile.build(ClassDesc.of("p", "ProviderFactory"), clb -> {
+            clb.withSuperclass(CD_Object);
+            clb.withFlags(AccessFlag.PUBLIC, AccessFlag.SUPER);
+            var provider$1Desc = ClassDesc.of("p", "ProviderFactory$1");
```

Comment:
This is class descriptor for ProviderFactory$1, not "Provider" so maybe rename this to providerFactory1 or something a bit clearer.

Reply:
> This is class descriptor for ProviderFactory$1, not "Provider" so maybe rename this to providerFactory1 or something a bit clearer.

The updated version looks good. I assume you'll do a pass over the updated tests to bump their copyright date as this is the first change in 2023 for many of these tests.